### ISSUE TITLE
Flow won't execute with Python 3.7

### ISSUE THREAD

**Issue by Tostino** [PrefectHQ/prefect#6666]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Prefect documentation for this issue.
- [X] I checked that this issue is related to Prefect and not one of its dependencies.

### Bug summary

I am having (file permissions?) failures when trying to run on my deployed infrastructure rather than a local docker container for the exact same flows mentioned in: https://github.com/PrefectHQ/prefect/issues/6665
The source files exist in the specified deploy location, they work in my local docker container, they don't work on this VM.

### Reproduction

```python
Use Debian 10, try to run a local storage and a process, with the code deployed where specified in the deployment.
```

### Error

```
Sep  1 01:41:12 check-prefectagent01 prefect[14263]: 01:41:12.436 | INFO    | prefect.agent - Submitting flow run '49507f89-c9e5-49b5-8b28-4e522f9f4253'
Sep  1 01:41:14 check-prefectagent01 prefect[14263]: 01:41:14.575 | INFO    | prefect.infrastructure.process - Opening process 'abstract-roadrunner'...
Sep  1 01:41:14 check-prefectagent01 prefect[14263]: 01:41:14.585 | INFO    | prefect.agent - Completed submission of flow run '49507f89-c9e5-49b5-8b28-4e522f9f4253'
Sep  1 01:41:16 check-prefectagent01 prefect[14263]: 01:41:16.336 | ERROR   | Flow run 'abstract-roadrunner' - Flow could not be retrieved from deployment.
Sep  1 01:41:16 check-prefectagent01 prefect[14263]: Traceback (most recent call last):
Sep  1 01:41:16 check-prefectagent01 prefect[14263]:   File "/usr/local/lib/python3.7/dist-packages/prefect/engine.py", line 256, in retrieve_flow_then_begin_flow_run
Sep  1 01:41:16 check-prefectagent01 prefect[14263]:     flow = await load_flow_from_flow_run(flow_run, client=client)
Sep  1 01:41:16 check-prefectagent01 prefect[14263]:   File "/usr/local/lib/python3.7/dist-packages/prefect/client.py", line 103, in with_injected_client
Sep  1 01:41:16 check-prefectagent01 prefect[14263]:     return await fn(*args, **kwargs)
Sep  1 01:41:16 check-prefectagent01 prefect[14263]:   File "/usr/local/lib/python3.7/dist-packages/prefect/deployments.py", line 54, in load_flow_from_flow_run
Sep  1 01:41:16 check-prefectagent01 prefect[14263]:     await storage_block.get_directory(from_path=deployment.path, local_path=".")
Sep  1 01:41:16 check-prefectagent01 prefect[14263]:   File "/usr/local/lib/python3.7/dist-packages/prefect/filesystems.py", line 132, in get_directory
Sep  1 01:41:16 check-prefectagent01 prefect[14263]:     shutil.copytree(from_path, local_path)
Sep  1 01:41:16 check-prefectagent01 prefect[14263]:   File "/usr/lib/python3.7/shutil.py", line 321, in copytree
Sep  1 01:41:16 check-prefectagent01 prefect[14263]:     os.makedirs(dst)
Sep  1 01:41:16 check-prefectagent01 prefect[14263]:   File "/usr/lib/python3.7/os.py", line 221, in makedirs
Sep  1 01:41:16 check-prefectagent01 prefect[14263]:     mkdir(name, mode)
Sep  1 01:41:16 check-prefectagent01 prefect[14263]: FileExistsError: [Errno 17] File exists: '.'
Sep  1 01:41:16 check-prefectagent01 prefect[14263]: 01:41:16.648 | INFO    | prefect.infrastructure.process - Process 'abstract-roadrunner' exited cleanly.

```

### Versions

```
$ prefect version
Version:             2.3.0
API version:         0.8.0
Python version:      3.7.3
Git commit:          8d9316c0
Built:               Tue, Aug 30, 2022 5:30 PM
OS/Arch:             linux/x86_64
Profile:             default
Server type:         hosted

```

### Additional context

```
[Adam Brusselback](https://app.slack.com/team/U040E5HMAMQ)
  [11 hours ago](https://prefect-community.slack.com/archives/CL09KU1K7/p1661997142018009?thread_ts=1661996757.734669&cid=CL09KU1K7)
Looks like the difference is my VM that is having the issue is running Debian 10 and Python 3.7, and the Docker container is running Python 3.9. (edited) 


[Adam Brusselback](https://app.slack.com/team/U040E5HMAMQ)
  [11 hours ago](https://prefect-community.slack.com/archives/CL09KU1K7/p1661997219571929?thread_ts=1661996757.734669&cid=CL09KU1K7)
I then saw this in the filesystems.py get_directory() function as a difference:
if sys.version_info < (3, 8):
    shutil.copytree(from_path, local_path)
else:
    shutil.copytree(from_path, local_path, dirs_exist_ok=True)
Also sent to the channel


[Adam Brusselback](https://app.slack.com/team/U040E5HMAMQ)
  [11 hours ago](https://prefect-community.slack.com/archives/CL09KU1K7/p1661998408397859?thread_ts=1661996757.734669&cid=CL09KU1K7)
And looking further, the docs say I should be able to set ignore_storage=True and things should be fine... that's not possible from the code I see though: https://docs.prefect.io/api-ref/prefect/deployments/?h=#prefect.deployments.load_flow_from_flow_run
[Jeff Hale](https://app.slack.com/team/U03KWSKFN8L)
:prefect:  [42 minutes ago](https://prefect-community.slack.com/archives/CL09KU1K7/p1662033913245079?thread_ts=1661996757.734669&cid=CL09KU1K7)
Hi Adam. Thanks for the info. Sounds like you might have found a bug. I looked at the issues and didn’t see an open one on the topic. Are you able to open an issue with an MRE over at https://github.com/PrefectHQ/prefect/issues/6641.
```

**Comment by madkinsz** [comment]
Thanks for the issue. Our difference in `filesystems.py` seems quite important so I'll highlight it again here.

```python
if sys.version_info < (3, 8):
    shutil.copytree(from_path, local_path)
else:
    shutil.copytree(from_path, local_path, dirs_exist_ok=True)
```

Is this a Python 3.7 debug or a Debian bug?

**Comment by davidhlavaty** [comment]
@madkinsz this is a bug  in Prefect code when Python 3.7 and lower is used. `local_path` defaults to current working directory. That will always exists (by definition), yet `shutil.copytree` on Python 3.7 and lower fails if the dest directory already exists ([docs](https://docs.python.org/3.7/library/shutil.html#shutil.copytree))

**Comment by madkinsz** [comment]
Yep. Can you update the title of your issue to reflect that this is an issue when using Prefect with Python 3.7 and not Debian 10?

**Comment by davidhlavaty** [comment]
There is another subtle bug - `shutil.copytree` will fail if `from_path` and `local_path` are the same (even on modern version of Python)

**Comment by Tostino** [comment]
Updated the issue title to better reflect the actual problem. I'm not a python dev, so thanks @davidhlavaty for the exact reason for the failure.

**Comment by madkinsz** [comment]
🤦 I didn't even realize it was a separate person. Thanks for the teamwork ya'll! We'll happily accept contributions to fix this, otherwise we'll get to it soon.

**Comment by vipulvisawadia** [comment]
@madkinsz We had a similar issue with Python 3.7 so wondering if this is going to be fixed soon or if we should move to Python 3.8

**Comment by madkinsz** [comment]
This is a high priority issue, but there is not an engineer assigned to it yet. I cannot give more details on the timeline, sorry. I would generally recommend moving to 3.8 anyway as 3.7 support was dropped from the Python scientific community (following https://numpy.org/neps/nep-0029-deprecation_policy.html#support-table) at the end of 2021.
